File: app/src/main/java/fr/free/nrw/commons/bookmarks/Bookmark.java
package fr.free.nrw.commons.bookmarks;

import android.net.Uri;

public class Bookmark {
    private Uri contentUri;
    private String mediaName;
    private String mediaCreator;

    public Bookmark(String mediaName, String mediaCreator, Uri contentUri) {
        this.contentUri = contentUri;
        this.mediaName = mediaName == null ? "" : mediaName;
        this.mediaCreator = mediaCreator == null ? "" : mediaCreator;
    }

    /**
     * Gets the media name
     * @return the media name
     */
    public String getMediaName() {
        return mediaName;
    }


    /**
     * Gets media creator
     * @return creator name
     */
    public String getMediaCreator() { return mediaCreator;  }



    /**
     * Gets the content URI for this bookmark
     * @return content URI
     */
    public Uri getContentUri() {
        return contentUri;
    }

}

File: app/src/main/java/fr/free/nrw/commons/bookmarks/Bookmark.kt
package fr.free.nrw.commons.bookmarks

import android.net.Uri

class Bookmark(mediaName: String?, mediaAuthor: String?,
               /**
                * Modifies the content URI - marking this bookmark as already saved in the database
                * @param contentUri the content URI
                */
               var contentUri: Uri?) {
    /**
     * Gets the content URI for this bookmark
     * @return content URI
     */
    /**
     * Gets the media name
     * @return the media name
     */
    val mediaName: String = mediaName ?: ""
    /**
     * Gets media author
     * @return author name
     */
    val mediaAuthor: String = mediaAuthor ?: ""

}